aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1498,39 +1498,49 @@ QuantizerPtr create_subtensor_quantizer(const Tensor& self, bool is_select, int6
   return quantizer;
 }
 
-Tensor select(const Tensor& self, int64_t dim, int64_t index) {
+Tensor select(const Tensor& self, int64_t dim, int64_t index_) {
   int64_t ndim = self.dim();
   if (ndim == 0) {
     TORCH_CHECK_INDEX(false, "select() cannot be applied to a 0-dim tensor.");
   }
   dim = maybe_wrap_dim(dim, ndim);
-  auto size = self.size(dim);
-  if (index < -size || index >= size) {
+  auto size = self.sym_sizes()[dim];
+  if (size < -index_ || size <= index_) {
```

Comment:
Did this condition change?

Reply:
No it did not. But the SymInt has to be on the left side and cannot have `-SymInt`. So just moved stuff around :)

Reply:
should just add enough overloads to SymInt.h so that we don't have to do this...

Reply:
What is the overload that should be implemented for `int64_t < SymInt` ?